### PR TITLE
Use reflection to access attach API internal APIs

### PR DESCRIPTION
**Pull request by pdbain-ibm**
Remove unnecessary compiler options related to exposing the APIs.

Fixes 
https://github.com/eclipse/openj9/issues/2938.

@llxia I ran a personal build with sanity tests on Java 10.  Would you kindly start builds to cover extended and Java 8.  Thank you.

Signed-off-by: Peter Bain <peter_bain@ca.ibm.com>

### CODE REVIEW

**Comment by llxia** [comment]
Jenkins test all plinux all

**Comment by llxia** [comment]
Jenkins test sanity xlinux jdk11

**Comment by pdbain-ibm** [comment]
This needs more work:
```
===============================================
Running test AttachAPISanity_SE80_0 ...
===============================================
AttachAPISanity_SE80_0 Start Time: Thu Sep 20 16:50:02 2018 Epoch Time (ms): 1537462202936
test with NoOptions
[IncludeExcludeTestAnnotationTransformer] [INFO] exclude file is /home/jenkins/jenkins-agent/workspace/PullRequest-Sanity-JDK8-linux_ppc-64_cmprssptrs_le-OpenJ9/openj9/test/TestConfig/../../jvmtest/TestConfig/resources/excludes/latest_exclude_8.txt
...
... TestNG 6.14.2 by Cédric Beust (cedric@beust.com)
...

[TargetManager] [ERROR] starting

[TargetManager] [ERROR] 0 failed

[TargetManager] [ERROR] Throwable: com.ibm.tools.attach.target.AttachHandler.getVmId(java.lang.String)

[TargetManager] [ERROR] java.lang.NoSuchMethodException: com.ibm.tools.attach.target.AttachHandler.getVmId(java.lang.String)

[TargetManager] [ERROR] 	at java.lang.Class.newNoSuchMethodException(Class.java:562)

[TargetManager] [ERROR] 	at java.lang.Class.throwExceptionOrReturnNull(Class.java:1195)

[TargetManager] [ERROR] 	at java.lang.Class.getMethodHelper(Class.java:1259)

[TargetManager] [ERROR] 	at java.lang.Class.getMethod(Class.java:1187)

[TargetManager] [ERROR] 	at org.openj9.test.attachAPI.TargetVM.main(TargetVM.java:46)

[TargetManager] [ERROR] starting

[TargetManager] [ERROR] 0 failed

[TargetManager] [ERROR] Throwable: com.ibm.tools.attach.target.AttachHandler.getVmId(java.lang.String)

[TargetManager] [ERROR] java.lang.NoSuchMethodException: com.ibm.tools.attach.target.AttachHandler.getVmId(java.lang.String)
```

**Comment by pdbain-ibm** [comment]
Fixed and re-tested.  @llxia would you kindly run sanity and extended builds on whichever platform is convenient?
   Thank you.

File: test/functional/TestUtilities/.settings/org.eclipse.jdt.core.prefs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+org.eclipse.jdt.core.compiler.codegen.methodParameters=do not generate
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
+org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
+org.eclipse.jdt.core.compiler.compliance=1.8
+org.eclipse.jdt.core.compiler.debug.lineNumber=generate
+org.eclipse.jdt.core.compiler.debug.localVariable=generate
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
Comment (llxia) [inline test/functional/TestUtilities/.settings/org.eclipse.jdt.core.prefs]: Do we need to check in this file?

Comment (pdbain-ibm) [reply test/functional/TestUtilities/.settings/org.eclipse.jdt.core.prefs]: No.  I included that by accident and have removed it.
Could we add `.settings` and other auto-generated files to the `.gitignore` file?

Comment (llxia) [reply test/functional/TestUtilities/.settings/org.eclipse.jdt.core.prefs]: yes, I think we can.

**Comment by llxia** [comment]
Jenkins test all plinux all

**Comment by llxia** [comment]
Jenkins test sanity win jdk11

**Pull request merged**